code/modules/mob/living/carbon/human/human.dm

```diff
@@ -52,15 +52,25 @@
 
 
 /mob/living/carbon/human/Dispose()
-	if(assigned_squad)
-		var/n = assigned_squad.marines_list.Find(src)
-		if(n)
-			assigned_squad.marines_list[n] = name //mob reference replaced by name string
-		if(assigned_squad.squad_leader == src)
-			assigned_squad.squad_leader = null
-		assigned_squad = null
+	clean_from_squad(FALSE)
 	remove_from_all_mob_huds()
-	. = ..()
+	return ..()
+
+/mob/living/carbon/human/proc/clean_from_squad(wipe = FALSE)
+	if(!assigned_squad)
+		return FALSE
+	var/n = assigned_squad.marines_list.Find(src)
```

Comment:
Make marines_list an associative array in the format human:TRUE and do `assigned_squad.marines_list -= src` instead